hazelcast/src/main/java/com/hazelcast/kubernetes/KubernetesClient.java

```diff
@@ -706,27 +738,28 @@ final class StsMonitor implements Runnable {
          */
         @Override
         public void run() {
-            String stsUrlString = String.format("%s/apis/apps/v1/namespaces/%s/statefulsets", kubernetesMaster,
-                    namespace);
-            JsonObject jsonObject = callGet(stsUrlString);
-            latestResourceVersion = jsonObject.get("metadata").asObject().getString("resourceVersion",
-                    null);
-            latestRuntimeContext = extractStsList(jsonObject);
-            LOGGER.info("Initializing cluster topology tracker with initial context: "
-                    + latestRuntimeContext);
-            clusterTopologyIntentTracker.update(UNKNOWN,
-                    latestRuntimeContext.getSpecifiedReplicaCount(),
-                    UNKNOWN, latestRuntimeContext.getReadyReplicas(),
-                    UNKNOWN, latestRuntimeContext.getCurrentReplicas());
-            while (true) {
+            BackoffIdleStrategy backoffIdleStrategy = new BackoffIdleStrategy(3, MAX_YIELDS,
+                    MILLISECONDS.toNanos(1), SECONDS.toNanos(MAX_PARK_PERIOD_SECONDS));
+            RestClient.WatchResponse watchResponse;
+            String message;
+
+            while (running) {
                 if (Thread.interrupted()) {
                     break;
                 }
-                RestClient restClient = RestClient.create(stsUrlString)
-                        .withHeader("Authorization", String.format("Bearer %s", tokenProvider.getToken()))
-                        .withCaCertificates(caCertificate);
-                RestClient.WatchResponse watchResponse = restClient.watch(latestResourceVersion);
-                String message;
+                try {
+                    // read initial statefulset list
+                    RuntimeContext previous = latestRuntimeContext;
+                    readInitialStsList();
+                    // update tracker
+                    updateTracker(previous, latestRuntimeContext);
+                    watchResponse = sendWatchRequest();
+                } catch (RestClientException e) {
+                    handleFailure(backoffIdleStrategy, e);
+                    continue;
```

Comment:
maybe a tiny comment before `continue` to indicate we always retry when there is a rest-client exception.

Reply:
35f52e7d2e5a57e6062dd409e88c557daf676752